packages/firebase_database/android/src/main/java/io/flutter/plugins/firebase/database/FirebaseDatabasePlugin.java

```diff
@@ -118,38 +118,101 @@ public void onDataChange(DataSnapshot snapshot) {
   @SuppressWarnings("unchecked")
   @Override
   public void onMethodCall(MethodCall call, final Result result) {
-    Map<String, ?> arguments = (Map<String, ?>) call.arguments;
-    if (call.method.equals("DatabaseReference#set")) {
-      Object value = arguments.get("value");
-      getReference(arguments).setValue(value, new DefaultCompletionListener(result));
-    } else if (call.method.equals("Query#observe")) {
-      String eventType = (String) arguments.get("eventType");
-      int handle = nextHandle++;
-      EventObserver observer = new EventObserver(eventType, handle);
-      observers.put(handle, observer);
-      if (eventType.equals(EVENT_TYPE_VALUE)) {
-        getReference(arguments).addValueEventListener(observer);
-      } else {
-        getReference(arguments).addChildEventListener(observer);
-      }
-      result.success(handle);
-    } else if (call.method.equals("Query#removeObserver")) {
-      DatabaseReference reference = getReference(arguments);
-      int handle = (Integer) arguments.get("handle");
-      EventObserver observer = observers.get(handle);
-      if (observer != null) {
-        if (observer.requestedEventType.equals(EVENT_TYPE_VALUE)) {
-          reference.removeEventListener((ValueEventListener) observer);
-        } else {
-          reference.removeEventListener((ChildEventListener) observer);
+    Map<String, Object> arguments = (Map<String, Object>) call.arguments;
+    switch (call.method) {
+      case "FirebaseDatabase#goOnline":
+        {
```

Comment:
It doesn't, I was trying to be consistent with the others but maybe it's better not to do a block. I've removed the braces for the methods that don't take arguments.